pulp_ansible/app/tasks/collections.py

```diff
@@ -120,47 +122,24 @@ def import_collection(
     CollectionImport.objects.get_or_create(task_id=get_current_job().id)
 
     artifact = Artifact.objects.get(pk=artifact_pk)
-    log.info("Processing collection from {path}".format(path=artifact.file.path))
+    filename = CollectionFilename(expected_namespace, expected_name, expected_version)
+    log.info(f"Processing collection from {artifact.file.name}")
     import_logger = logging.getLogger("pulp_ansible.app.tasks.collection.import_collection")
 
-    try:
-        importer_result = process_collection(str(artifact.file.path), logger=import_logger)
-        collection_info = importer_result["metadata"]
-
-        if expected_namespace and expected_namespace != collection_info["namespace"]:
-            msg = _(
-                "Expected namespace '{expected_namespace}', but instead Collection has namespace "
-                "'{collection_namespace}'."
-            )
-            raise ValidationError(
-                msg.format(
-                    expected_namespace=expected_namespace,
-                    collection_namespace=collection_info["namespace"],
+    with _artifact_guard(artifact):
+        try:
+            with artifact.file.open() as artifact_file:
+                importer_result = process_collection(
+                    artifact_file, filename=filename, logger=import_logger
                 )
-            )
 
-        if expected_name and expected_name != collection_info["name"]:
-            msg = _(
-                "Expected name '{expected_name}', but instead Collection has name "
-                "'{collection_name}'."
-            )
-            raise ValidationError(
-                msg.format(expected_name=expected_name, collection_name=collection_info["name"])
-            )
+        except ImporterError as exc:
+            # NOTE(cutwater): Because this one is expected, probably should wrap into some
```

Comment:
With the future pulp_ansible work tracked in Redmine, can we convert things like this into Remdine tickets instead of here? I strive to keep planning in one place.

Reply:
It's more like a question, I'm not sure if this has to be fixed or not.

Reply:
oic. In that case I would say let's not fix it for now and not file it as future work. We can improve this later when we get a bugfix. What do you think?

Reply:
So what do I do with this comment? Remove it?

Reply:
+1 to removing the comment yes.

Reply:
Done